Fixes CMSIS v2 test on x86

dcpleung opened the pull request:
* In test_event_flags_no_wait_timeout(), after creating thread1, there is a little delay to let it run to trigger FLAG1. However, in test_event_flags_signalled(), this is not being done, and on some platforms it triggers the assert in thread1() complaining the flag not being set. This adds the same delay in test_event_flags_signalled() and the this test passes for those previously failed platforms.
* Some of the thread tests are hard-coded with assumption that there is only one CPU. So limit the number of CPUs to 1 via kconfig.
* The conditions which prohibit running the test suite on qemu_x86_64 and up_squared no longer exist. So remove these two from the exclude list, and now they can be built and run in CI.

Fixes #25507

dcpleung committed:
tests: cmsis_rtos_v2: wait a bit for thread1 to set flags

In test_event_flags_no_wait_timeout(), after creating thread1,
there is a little delay to let it run to trigger FLAG1.
However, in test_event_flags_signalled(), this is not being
done, and on some platforms it triggers the assert in thread1()
complaining the flag not being set. This adds the same delay
in test_event_flags_signalled() and the this test passes for
those previously failed platforms.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
tests: cmsis_rtos_v2: limit number of CPUs to 1

Some of the thread tests are hard-coded with assumption that
there is only one CPU. So limit the number of CPUs to 1
via kconfig.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
tests: cmsis_rtos_v2: enable x86 64-bit boards

The conditions which prohibit running the test suite on
qemu_x86_64, up_squared and ehl_crb no longer exist. So
remove these two from the exclude list, and now they can
be built and run in CI.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

jhedberg commented:
@dcpleung this needs to be synced with #30606 since that one is adding another board to the exclude list. Ie whichever PR goes in last will need updating.

abrodkin commented:
I think @IRISZZW bumped into this as well and was going to suggest the same fix as in https://github.com/zephyrproject-rtos/zephyr/pull/30621/commits/65485c9b60b590af36e4cbdee40779614adbb9c9. So that's good it will be fixed for SMP systems now.

BTW @dcpleung it might make sense that at least https://github.com/zephyrproject-rtos/zephyr/pull/30621/commits/65485c9b60b590af36e4cbdee40779614adbb9c9 is needed for any SMP system.

nashif commented:
> @dcpleung this needs to be synced with #30606 since that one is adding another board to the exclude list. Ie whichever PR goes in last will need updating.

this might fix the issue on ehl as well, so no need to exclude.

nashif commented:
@dcpleung  this hangs for me on 

```
*** Booting Zephyr OS build zephyr-v2.4.0-2271-g89c456d897fc  ***
Running test suite test_cmsis_v2_apis
===================================================================
START - test_kernel_apis
 PASS - test_kernel_apis
===================================================================
START - test_delay
```

jhedberg commented:
> this might fix the issue on ehl as well, so no need to exclude.

I'm almost sure it does fix it, which was my point: if my PR gets merged first then this PR needs rebasing, and if this PR gets merged first then I need to update my PR to not try to add anything to the excludes.

dcpleung commented:
> @dcpleung this hangs for me on
> 
> ```
> *** Booting Zephyr OS build zephyr-v2.4.0-2271-g89c456d897fc  ***
> Running test suite test_cmsis_v2_apis
> ===================================================================
> START - test_kernel_apis
>  PASS - test_kernel_apis
> ===================================================================
> START - test_delay
> ```

Hm... ran a couple times on my boards back-to-back without any issues. Let me try run it more times and see if I can reproduce this.

dcpleung commented:
Ran 15 times back-to-back without hanging.

dcpleung commented:
Found the issue with hanging under `test_delay`. It is caused by the same issue as in #30574 on `CONFIG_INTEL_VTD_ICTL` being enabled.

jhedberg commented:
@dcpleung #30606 is now merged so you'll need to rebase and fix the test conflict in test case.yaml. You can just remove the ehl_crb boad like you're doing with up_squared.

dcpleung commented:
> @dcpleung #30606 is now merged so you'll need to rebase and fix the conflict in testcase.yaml. You can just remove the ehl_crb boad like you're doing with up_squared.

Thanks for letting me know. Rebased and removed all x86 64-bit boards from the exclude list.

chen-png reviewed (approved):
tested and it fixed 25507
